src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.cpp

```diff
@@ -0,0 +1,570 @@
+    {
+        args.iceTransportPolicy.SetValue(std::string(req.ICETransportPolicy.Value().data(), req.ICETransportPolicy.Value().size()));
+    }
+
+    // Delegate processing:
+    // The delegate implementation SHALL:
+    // - Populate a new WebRTCSessionStruct with the requested values.
+    // - If in standby mode, set deferredOffer to true (and perform steps to exit standby within 30 seconds).
+    // - If not in standby mode, ensure that VideoStreamID and AudioStreamID are valid (or allocate/select new streams if null).
+    // - Internally increment the ReferenceCount on any used video/audio streams.
+    // - If resources cannot be allocated, invoke End with Reason OutOfResources.
+    WebRTCSessionStruct outSession;
+    bool deferredOffer = false;
+
+    auto status = Protocols::InteractionModel::ClusterStatusCode(mDelegate.HandleSolicitOffer(args, outSession, deferredOffer));
+    if (!status.IsSuccess())
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+        return;
+    }
+
+    // Store or update the session.
+    if (UpsertSession(outSession) == UpsertResultEnum::kInserted)
+    {
+        ChipLogProgress(Zcl, "WebRTCTransportProvider: Inserted a new session with ID=%u.", outSession.id);
+    }
+    else
+    {
+        ChipLogProgress(Zcl, "WebRTCTransportProvider: Updated existing session with ID=%u.", outSession.id);
+    }
+
+    // Prepare the SolicitOfferResponse command.
+    Commands::SolicitOfferResponse::Type resp;
+    resp.webRTCSessionID = outSession.id;
+    resp.deferredOffer   = deferredOffer;
+
+    // If VideoStreamID was in the request, it should be in the response
+    if (req.videoStreamID.HasValue())
+    {
+        resp.videoStreamID.SetValue(outSession.videoStreamID);
+    }
+
+    // If AudioStreamID was in the request, it should be in the response
+    if (req.audioStreamID.HasValue())
+    {
+        resp.audioStreamID.SetValue(outSession.audioStreamID);
+    }
+
+    ctx.mCommandHandler.AddResponse(ctx.mRequestPath, resp);
+}
+
+void WebRTCTransportProviderServer::HandleProvideOffer(HandlerContext & ctx, const Commands::ProvideOffer::DecodableType & req)
+{
+    auto webRTCSessionID = req.webRTCSessionID;
+    auto videoStreamID   = req.videoStreamID;
+    auto audioStreamID   = req.audioStreamID;
+
+    NodeId peerNodeId = GetNodeIdFromCtx(ctx.mCommandHandler);
+
+    WebRTCSessionStruct outSession;
+
+    // New session request: webRTCSessionID is null.
+    if (webRTCSessionID.IsNull())
+    {
+        // If both video and audio stream IDs are missing, return CONSTRAINT_ERROR.
+        if (!videoStreamID.HasValue() && !audioStreamID.HasValue())
+        {
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::ConstraintError);
+            return;
+        }
+
+        // Validate the streamUsage field against the allowed enum values.
+        if (req.streamUsage == StreamUsageEnum::kUnknownEnumValue)
```

Comment:
This check is in the wrong place.  This needs to happen regardless of the value of the webRTCSessionID field, no?